### ISSUE TITLE
Selecting multiple categories in embedded reports results in blank exports

### ISSUE THREAD

**Issue by arcreative** [metabase/metabase#7976]
As titled, selecting _one_ category filter in embedded reports works just fine, but selecting _two or more_ categories shows the proper data in the table, but results in a blank export (just headers, no data) when saving results.  

Sample SQL query:
```
SELECT *
FROM feedback
WHERE true
  [[AND {{feedback_type}}]]
  [[AND {{state}}]]
  [[AND {{year}}]]
  [[AND {{half}}]]
  [[AND {{quarter}}]]
```

Details:
- Your browser and the version: 67.0.3396.99
- Your operating system: OSX High Sierra
- Your databases: Postgres
- Metabase version: v0.29.3
- Metabase hosting environment: Ubuntu Server 16.04
- Metabase internal database: Postgres

To reproduce
* Create a table of raw data based on the SQL snippet above
* Make the filters "category" type based on the field in question.  The field settings are set to "show all the values"
* Embed the report in another page
* Set two values for the same field (any of them)
* Click export and note that export does not show any data

This is pretty crucial functionality for us, so would be great to have some information about how to fix it!

**Comment by arcreative** [comment]
@salsakran Sorry for not being familiar with your team's approach to bugs and updates, but is there some sort of ballpark estimate for when this might be fixed or released?  This is pretty crippling for my team at the moment.  I can put together a PR if that helps, I would just need some sort of idea of where to start!

**Comment by salsakran** [comment]
So realistically - we're a tiny team creating a product used by 10s of thousands of people in a staggering variety of configurations. We prioritize bugs based on severity of impact and how many people are impacted. So far this is a bug with a single reported occurrence, and we honestly can't give a timeline. 

I'll also note that we offer paid support for situations where individual companies need to get a specific bug fixed asap in situations where it wouldn't otherwise be prioritized highly.

**Comment by arcreative** [comment]
@salsakran Ok thanks—any pointers about how you would attack this?  I would love to give it a try, but this is a new language to me :-)

**Comment by jornh** [comment]
Hi @arcreative,

I’ll try to have a go at this. I’m *not* part of that tiny Metabase team - just another open source user happening to [spend](https://github.com/metabase/metabase/issues?utf8=✓&q=+is%3Aissue+commenter%3Ajornh+sort%3Areactions-%2B1-desc+) a lot of my personal [time](http://discourse.metabase.com/u/jornh/summary) on Metabase.

First of this could actually be a variation or even dupe of https://github.com/metabase/metabase/issues/5295 - with 20 👍 votes. If my assumption is right this suddenly rates popularity/impact higher. But no one of us here knew that just an hour before, right?

Second, by asking @salsakran to give those pointers on how to attack this, and not first demonstrating what effort you made towards at least step 1 below, you’re effectively trying to bump the priority queue he just mentioned. (I’ve by now spent at least ~~half~~ an hour just on triaging your issue, finding #5295  in that process, and responding to you). I think this is just the harsh realities of OSS.

Here my initial pointers for you on how I would try to squash this bug:

1. Start from the top - familiarize yourself with development: 
https://github.com/metabase/metabase/blob/master/docs/developers-guide.md

2. Add a test on the rest API exposing the bug. Could be around here:
https://github.com/metabase/metabase/blob/v0.29.3/test/metabase/api/card_test.clj#L683
(That’s me assuming the bug is in the backend code, if that turns out to not be true, you already have narrowed the bug down a bit) 

3. Make the test green again :-)

**Comment by arcreative** [comment]
@jornh Thanks for your comment.  I’m not trying to skirt the issue and bump priority, and I have spent more than zero time in the code trying to track it down.  It’s just that those familiar with the project can probably point me to a class (say, a parameter parser, not HOW one contributes to OSS) where the issue is likely to occur, which could really help me help them.  Apologies if it came off as whining like most people tend to do on GitHub, but I’m quite familiar with OSS and TDD, was just looking to save a little time.

Will take a look at this as well.

**Comment by jornh** [comment]
Good to hear. All fine and well then, no worries!

If you're willing to put in the work I guess anything's possible. More than happy to as best I can to try to answer more specific questions, when you get to em'. The thing I get out of that is that I also get to learn a bit more Clojure, this codebase etc.

**Comment by arcreative** [comment]
After performing an upgrade, this seems to be fixed in 0.30.0 (lucky it just came out!).  Thanks all.

**Comment by jornh** [comment]
Have to quote myself there:

> I guess anything's possible
